File: backend/app/common/dto/DataRequestDto.py
from pathlib import Path
from typing import Union, BinaryIO

from pydantic import BaseModel, Field, ConfigDict

class DataRequestDto(BaseModel):
    model_config = ConfigDict(arbitrary_types_allowed=True)

    l2: float
    k1: float
    l1: float
    alfa1: float
    alfa2: float
    alfa3: float
    r21: float

    sample_name: str
    use_hankel: bool
    file: Union[str, Path, BinaryIO]

File: frontend/src/models/data-request-dto.ts
export interface DataRequestDto {
    l2: number;
    k1: number;
    l1: number;
    alfa1: number;
    alfa2: number;
    alfa3: number;
    r21: number;
    weight: number;

    sample_name: string;
    use_hankel: boolean;
    file: File;
}
